python/src/aiconfig/util/config_utils.py

```diff
@@ -10,10 +11,28 @@
     from ..schema import AIConfig
 
 
-def get_api_key_from_environment(api_key_name: str):
-    if api_key_name not in os.environ:
-        raise Exception("Missing API key '{}' in environment".format(api_key_name))
+def maybe_get_api_key_from_environment(
+    api_key_name: str,
+    required: bool = True) -> Union[str, None]:
+    """Get the API key if it exists, return None or error if it doesn't
+
+    Args:
+        api_key_name (str): The keyname that we're trying to import from env variable
+        required (bool, optional): If this is true, we raise an error if the 
+            key is not found
+
+    Returns:
+        Union[str, None]: the value of the key. If `required` is false, this can be None
+    """
+    dotenv.load_dotenv()
```

Comment:
Nice, I always thought it was strange that we had to manually use dotenv everywhere instead of just having it done once here. Is there a reason this wouldn't be wanted?

Reply:
Wouldn't this load dotenv every time this function is called? Is that what we want?

Reply:
> Wouldn't this load dotenv every time this function is called?

Even if we did call this every time, I don't think it's an issue?

This is only called on the `init()` for the hugging face model parsers, and only if the api key is not set in OpenAI (which it will be after calling this function once): https://github.com/search?q=repo%3Alastmile-ai%2Faiconfig%20get_api_key_from_environment&type=code

I'll edit the Gemini callsite in this diff to only call this one time just like OpenAI: https://github.com/lastmile-ai/aiconfig/blob/5b28259ff44dd3afd03c40206888ec06774ccaed/extensions/Gemini/python/src/aiconfig_extension_gemini/Gemini.py#L329

Reply:
^Gemini fixed in https://github.com/lastmile-ai/aiconfig/pull/786

Reply:
Ok sounds good, thanks!